apps/mobile/src/screens/chat/chat.tsx

```diff
@@ -1,22 +1,77 @@
-import React from "react";
-
-import { Checkbox, ScreenWrapper, View } from "~/libs/components/components";
-import { useState } from "~/libs/hooks/hooks";
+import {
+	CheckboxCategoriesForm,
+	ScreenWrapper,
+	ScrollView,
+	Text,
+	View,
+} from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import {
+	useAppDispatch,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useState,
+} from "~/libs/hooks/hooks";
 import { globalStyles } from "~/libs/styles/styles";
+import { type CategoriesSelectedRequestDto } from "~/libs/types/types";
+import { actions as categoriesActions } from "~/slices/quiz/quiz";
+
+import { ChatBox } from "./libs/components/components";
+import { styles } from "./styles";
 
 const Chat: React.FC = () => {
-	const [isChecked, setIsChecked] = useState<boolean>(false);
+	const [submittedCategoryIds, setSubmittedCategoryIds] = useState<
+		null | number[]
+	>(null);
+	const dispatch = useAppDispatch();
+	const categories = useAppSelector((state) => state.quiz.categories);
+
+	useEffect(() => {
+		void dispatch(categoriesActions.getScores());
+	}, [dispatch]);
+
+	const handleRetakeQuizSubmit = useCallback(
+		(payload: CategoriesSelectedRequestDto): void => {
+			setSubmittedCategoryIds(payload.categoryIds);
+			// TODO: send selectedCategoriesSubmissionData to backend
+		},
+		[],
+	);
 
 	return (
-		<ScreenWrapper>
-			<View style={[globalStyles.flex1, globalStyles.gap12, globalStyles.p12]}>
-				<Checkbox
-					isChecked={isChecked}
-					label="Physical"
-					onValueChange={setIsChecked}
-				/>
-			</View>
-		</ScreenWrapper>
+		<View style={[globalStyles.flex1, styles.container]}>
+			<ScreenWrapper>
```

Comment:
Ideally the `ScreenWrapper` should be a top wrapper for the component because it provides the safe edges. If you need to change the background color you can add a `style` prop directly to `ScreenWrapper`